rb/spec/integration/selenium/webdriver/key_input_spec.rb

```diff
@@ -0,0 +1,74 @@
+# encoding: utf-8
+#
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+require_relative 'spec_helper'
+
+module Selenium
+  module WebDriver
+    compliant_on driver: :firefox do
+      describe W3CActionBuilder do
+        describe 'KeyInput interactions' do
+          it 'sends keys to the active element' do
+            driver.navigate.to url_for('bodyTypingTest.html')
+            keylogger = driver.find_element(id: 'body_result')
+
+            driver.action.send_keys('ab').perform
+
+            wait.until { keylogger.text.split.size == 2 }
+            expect(keylogger.text.strip).to eq('keypress keypress')
+
+            expect(driver.find_element(id: 'result').text.strip).to be_empty
+          end
+
+          it 'can send keys with shift pressed' do
+            driver.navigate.to url_for('javascriptPage.html')
+
+            event_input = driver.find_element(id: 'theworks')
+            keylogger = driver.find_element(id: 'result')
+
+            event_input.click
+
+            driver.action.key_down(:shift).send_keys('ab').key_up(:shift).perform
+            wait.until { event_input.attribute(:value).size == 2 }
+
+            expect(event_input.attribute(:value)).to eq('AB')
+            expect(keylogger.text.strip).to match(/^(focus )?keydown keydown keypress keyup keydown keypress keyup keyup$/)
```

Comment:
if we're not capturing this regex, we can use `[]` as non-capturing items are "slightly" faster. But again it's probably negligible